airflow/task/task_runner/standard_task_runner.py

```diff
@@ -17,28 +17,69 @@
 # specific language governing permissions and limitations
 # under the License.
 
+import os
+
 import psutil
+from setproctitle import setproctitle
 
 from airflow.task.task_runner.base_task_runner import BaseTaskRunner
 from airflow.utils.helpers import reap_process_group
 
+CAN_FORK = hasattr(os, 'fork')
+
 
 class StandardTaskRunner(BaseTaskRunner):
     """
     Runs the raw Airflow task by invoking through the Bash shell.
     """
     def __init__(self, local_task_job):
         super().__init__(local_task_job)
+        self._rc = None
 
     def start(self):
-        self.process = self.run_command()
+        if CAN_FORK and not self.run_as_user:
+            self.process = self._start_by_fork()
+        else:
+            self.process = self._start_by_exec()
 
-    def return_code(self):
-        return self.process.poll()
+    def _start_by_exec(self):
+        subprocess = self.run_command()
+        return psutil.Process(subprocess.pid)
 
-    def terminate(self):
-        if self.process and psutil.pid_exists(self.process.pid):
-            reap_process_group(self.process.pid, self.log)
+    def _start_by_fork(self):
+        pid = os.fork()
+        if pid:
```

Comment:
Just wondering if we want to make it explicit. Maybe not required by just a suggestion:

```
if pid == 0:
```

example:


```python
    def _start_by_fork(self):
        pid = os.fork()
        if pid == 0:
            from airflow.bin.cli import CLIFactory
            parser = CLIFactory.get_parser()
            args = parser.parse_args(self._command[1:])
            setproctitle(
                "airflow task runner: {0.dag_id} {0.task_id} {0.execution_date} {0.job_id}".format(args)
            )
            args.func(args)
            os._exit(0)

        else:
            self.log.info("Started process %d to run task", pid)
            return psutil.Process(pid)
```

Reply:
Could do, but for python it doesn't matter so much - either it turns a pid, 0 in the child or throws an error (C can return 0, pid or -1 on error, but python converts that to an exception for us)